index.js

```diff
@@ -200,13 +200,19 @@ ASS.prototype.getUrl = function (endpoint, http) {
  * Create a signed URL from image id and actions.
  * @param  {Integer} id - image id
  * @param  {Object} [actions]
+ * @param  {Object} [options]
+ * @param  {Boolean} [options.https] if we should return https
+ * @param  {Boolean} [options.encode] if we should url encode the actions
  * @return {String}
  */
-ASS.prototype.createImageUrl = function (id, actions) {
-	var url = this.getUrl(printf('/users/%s/images/%s.jpg', this.username, id), true);
+ASS.prototype.createImageUrl = function (id, actions, options) {
+	options = options ? options : {};
+	options.https = typeof options.https === 'boolean' ? options.https : false;
+	options.encode = typeof options.encode === 'boolean' ? options.encode : false;
+	var url = this.getUrl(printf('/users/%s/images/%s.jpg', this.username, id), !options.https);
 
 	if (actions) {
-		url += '?' + qs.stringify({ t: actions }, { encode: false });
+		url += '?' + qs.stringify({ t: actions }, { encode: options.encode });
```

Comment:
Yeah, realized that so rewrote it to create the signature based on the un-encoded url. Worked fine now when I tested it locally.